File: php/features.php
<?php
declare(strict_types=1);

// StoicismQuote SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class StoicismQuoteFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new StoicismQuoteBaseFeature();
            case "test":
                return new StoicismQuoteTestFeature();
            default:
                return new StoicismQuoteBaseFeature();
        }
    }
}

File: py/features.py
# StoicismQuote SDK feature factory

from feature.base_feature import StoicismQuoteBaseFeature
from feature.test_feature import StoicismQuoteTestFeature


def _make_feature(name):
    features = {
        "base": lambda: StoicismQuoteBaseFeature(),
        "test": lambda: StoicismQuoteTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
